### ISSUE TITLE
[SR-1147] Optional capture specifiers misleading in capture lists

### ISSUE THREAD

**Issue by swift-ci** [swiftlang/swift#43760]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-1147      |
   |Radar             | None         |
   |Original Reporter | zoul (JIRA User)      |
   |Type              | Improvement    |

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | Compiler    |
   |Labels            | Improvement, LanguageFeatureRequest        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: f79e9c82a44f19decd139670bdc17593

  </details>

**Issue Description:**

I have the following capture list:

``` swift
let closure = { [unowned x, y] in … }
```

Now `x` is `unowned`, but it’s quite reasonable to expect `y` to be `unowned`, too. This doesn’t seem to be the case:

``` swift
class Test {

    var callback: (Void -> Void)?

    init() {
        print("init")
    }

    deinit {
        print("deinit")
    }
}

func makeScope() {
    let a, b: Int
    let x = Test()
    let y = Test()
    y.callback = { [unowned x, y] in
        print(y)
    }
}

makeScope()
print("done")
```

This prints:

``` java
init
init
deinit
done
```

In other words, `y` gets involved in a retain cycle and leaks. Misreading `unowned x, y` as `unowned x, unowned y` is quite an easy mistake to make, and unfortunately you don’t realize your error until you notice the leak. How about at least warning about missing capture specifiers in front of reference types?
